src/renderer/vt/paint.cpp

```diff
@@ -448,10 +448,13 @@ using namespace Microsoft::Console::Types;
     // the lines _wrapped_. It doesn't care to manually break the lines, but if
     // we trimmed the spaces off here, we'd print all the "~"s one after another
     // on the same line.
-    const auto removeSpaces = !lineWrapped && (useEraseChar ||
-                                               _clearedAllThisFrame ||
-                                               (_newBottomLine && printingBottomLine && bgMatched));
-    const auto cchActual = removeSpaces ? nonSpaceLength : cchLine;
+    static const TextAttribute defaultAttrs{};
+    const bool removeSpaces = !lineWrapped && (useEraseChar // we determined earlier that ECH is optimal
+                                               || (_clearedAllThisFrame && _lastTextAttributes == defaultAttrs) // OR we cleared the last frame to the default attributes (specifically)

```

Comment:
I don't think I ever recommended this on account of its fragility, and I meant to ask @j4james if this seemed ... uh, reasonable?

Sorry, I'm still a bit brain-fogged

Reply:
HONESTLY I'm just opening the pr because I was in the area. I tried it for #13229, and it didn't work for that. However, it seemed overall reasonable, so filed this for discussion to get some feedback, mostly from @j4james.

Reply:
I don't know know the conpty code very well, so I'm unsure if it's guaranteed that `_clearedAllThisFrame` would mean the screen was cleared with the default attributes. Is that correct? It's not possible for conpty to inherit colors from somewhere in the first frame, like maybe when nesting conpty sessions?

But if you're sure the screen was cleared with default attributes, and we're now writing out text with default attributes, then I think that would imply it's safe to remove spaces.

Reply:
Looks like the only place `_clearedAllThisFrame = true` is set is in the first paint of `StartPaint`, when yea, the screen was cleared with the defaults. I don't think nesting conpty would change that - even if you did `color 46`, `wsl`, `cmd.exe` in a Terminal cmd instance, that second conpty would spawn with default attrs. 

That sounds good enough to me!